core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -619,11 +620,16 @@ protected Type visitDereferenceExpression(DereferenceExpression node, StackableA
             RowType rowType = (RowType) baseType;
             String fieldName = node.getField().getValue();
 
+            int index = -1;
```

Comment:
Use `boolean found` instead of the integer.

Reply:
Copy that.